src/Compilers/Core/Portable/MetadataReference/AssemblyIdentityMap.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Implements a map from an assembly identity to a value. The map allows to look up the value by an identity
+    /// that either exactly matches the original identity key or whose version is lower (that is the resulting value 
```

Comment:
The following phrase feels confusing: "either exactly matches the original identity key **or whose version is lower**." Perhaps it should say "either exactly matches the original identity key, or corresponds to a key with the lowest version among identities with higher version than the requested identity key"?